src/http/request.rs

```diff
@@ -111,6 +111,15 @@ impl Request {
         self.0.connection
     }
 
+    /// Pointer to a [`ngx_http_upstream_t`] upstream server object.
+    ///
+    /// [`ngx_http_upstream_t`]: is best described in
+    /// https://nginx.org/en/docs/dev/development_guide.html#http_request
+    /// https://nginx.org/en/docs/dev/development_guide.html#http_load_balancing
+    pub fn upstream(&self) -> *mut ngx_http_upstream_t {
```

Comment:
Most of the `Request` functionality is somewhat safe and returns "wrapped" types (see pool, status, NgxStr, etc). let's aim that upstream is following this, otherwise, this is just writing C in Rust.

I can offer rename:
`pub fn upstream_ptr(&self) -> *mut ngx_http_upstream_t {`